masu/processor/providers/aws_report_processor.py

```diff
@@ -30,7 +30,7 @@
 from masu.database.report_db_accessor import ReportDBAccessor
 from masu.database.reporting_common_db_accessor import ReportingCommonDBAccessor
 from masu.exceptions import MasuProcessingError
-from masu.external import GZIP_COMPRESSED, UNCOMPRESSED
+from masu.external import GZIP_COMPRESSED
```

Comment:
I sort of like keeping all compression types here just so it's clear how each type is being opened and if a foreign compression type is mistakenly passed in then it would be implicitly opening with the wrong type.  For that case it would simply not open the file and we would be forced to look into what happened.  All that being said, we are in complete control of when this gets called, so it's probably no problem changing it.

Reply:
In general, the "pythonic" way tends to be to favor simplicity over unnecessary explicitness. That's why pylint complains when you do "if-else" instead of just "if with an implied else", like here.

Preferred:
```
if some-thing:
    do_thing()
do_other_thing()
```

Not preferred:
```
if some-thing:
    do_thing()
else:
    do_other_thing()
```

Functionally, they result in the same thing most of the time, and when the results differ it's often because of other assumptions about what the desired behavior is. So, it's forcing your hand here to re-think what you're really wanting this code to do. (As you accurately observed.)

I think in this particular case, if we wanted to raise an error on an unknown compression type, we should do that explicitly, rather than letting that be a side-effect of falling through the if-clause. 

It's technically true that if we don't know the file is a specific compression, we just have to assume it's uncompressed and wait for the parser failure to surface when the Masu worker tries to read the file. Without doing any additional analysis on the file, we simply can't know at this point in the code whether the file is truly readable.